Submitty/Submitty
# [Feature:System] Stop the worker if it has no space

### Please check if the PR fulfills these requirements:


### What is the current behavior?

This PR closes #7315, closes #7744, closes #7900

According to the [original issue](../issues/7315), the worker will get stuck if there is no space left.  And running [`restart_all...py`](../blob/main/sbin/restart_shipper_and_all_workers.py) script will not produce errors.

### What is the new behavior?

![Before_Restarting](https://user-images.githubusercontent.com/11363469/171321650-35e12346-6056-4f52-bdcb-57b884f4d8e6.png)
![Restarting](https://user-images.githubusercontent.com/11363469/171321773-0914e99b-ebe6-4f78-8169-30cb6a05721a.png)
![After_Restarting](https://user-images.githubusercontent.com/11363469/171321954-03e69876-7768-40e0-b032-69383ef45609.png)

The worker will check the disk usage when it launches.  The thresholds for Warning and Error are 70% and 93%, respectively.  If the worker's root partition usage meets the thresholds, it will

- leave a warning message on the log file;
- exit, if the usage meets the error threshold.

The [`restart_all...py`](../blob/main/sbin/restart_shipper_and_all_workers.py) script will verify the status of the service after restarting.  If some workers report `inactive`, then it displays a message.  


### Other information?
- Daemon Job Dispatcher
  A bash script could create the daemon job in the CLI, not through the PHP pages.

- Testing
  The testing could be done with a worker machine, and `dd` or `lvreduce` commands.  To fill up the worker machine,
  1. Run `df -h` to retrieve the current usage;
  2. Run `dd if=/dev/zero of=awesomeBigFile bs=1G count=n` to fill the disk with an `n` GiB file;
  3. Alternatively, run `lvreduce -r -L -${n}G /dev/mapper/${AWESOME_LVM_VOLUME}` to reduce the LVM volume by `${n}` GB, make sure to substitute `${AWESOME_LVM_VOLUME}` to your actual volume;
  4. Run `python3 ${SUBMITTY_INSTALL_DIR}/sbin/restart_shipper_and_all_workers.py` on the primary machine.

  Note that there is a difference between `df -h` and `shutil.disk_usage('/')`.  On my end, `df -h` percentage is greater than the latter one.
